### PR TITLE
[17.0][MIG] mail_tracking: Migration to 17.0

### PR DESCRIPTION
**Pull request by chienandalu**
Ongoing tasks:

- [x] Refactor webclient
- [x] Fix server logic
- TODO: Add more reactivity

This migration is ready for review. :tada: 

- You could still find tiny issues regarding the reactivity of the components, mainly when we set a failed message as reviewed in the search failed panel or in Discuss. We'll work on that.
- A deep refactor of the webclient code has been made in core and thus a complete refactor of the webclient side for `mail_tracking` was needed.
- We've been splitting the funcionality in component that can be esasily used across the core components, so hopefully everything is more modular now.
- We're sitting on top of some new core features as well. Like a message search component to filter the failed messages.
- Also regular server logic stuff was necessary, although it wasn't needed such a deep refactor as in the client side. Note that the tests are almost untouched in this part.

image

image

image

cc @Tecnativa TT49060

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash a bit administrative commits.

**Comment by Bart-dh** [comment]
Hi @chienandalu is this safe to use for production environments already?

**Comment by chienandalu** [comment]
> Hi @chienandalu is this safe to use for production environments already?

It should be ready by now.

File: mail_tracking/__manifest__.py [hunk 1]
```diff
@@ -26,33 +26,16 @@
...
+            "mail_tracking/static/src/core/discuss/*",
+            "mail_tracking/static/src/services/*",
+            "mail_tracking/static/src/components/message_tracking/*",
+            "mail_tracking/static/src/components/failed_message/*",
+            "mail_tracking/static/src/components/failed_message_review/*",
+            "mail_tracking/static/src/components/failed_messages_panel/*",
         ],
     },
-    "demo": ["demo/demo.xml"],
+    # "demo": ["demo/demo.xml"],
```
Comment (pedrobaeza) [inline mail_tracking/__manifest__.py]: Please re-add demo data failed messages for showcasing/testing them.

File: mail_tracking/controllers/mailbox.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+from odoo.addons.mail.controllers.mailbox import MailboxController
+
+
+class MailTrackingMailBoxController(MailboxController):
+    @route("/mail/failed/messages", methods=["POST"], type="json", auth="user")
+    def discuss_failed_messages(
+        self, search_term=None, before=None, after=None, limit=30, around=None
+    ):
+        """Fetch failed massages for discuss"""
```
Comment (pedrobaeza) [inline mail_tracking/controllers/mailbox.py]: What you were thinking in? :stuck_out_tongue: 

```suggestion
        """Fetch failed messages for discuss"""
```

File: mail_tracking/static/description/icon.png
Comment (pedrobaeza) [inline mail_tracking/static/description/icon.png:1]: Great new icon! Did you do it? If not, please include the credits.

Comment (chienandalu) [reply mail_tracking/static/description/icon.png:1]: Thanks! :smile: 

Yes, I took the Odoo mass_mailing icon and a font-awesome glyph and playing with the blending mode in inkscape

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**